test/SlowTests/Server/Documents/PeriodicBackup/Ftp.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentFTP;
+using FubarDev.FtpServer;
+using FubarDev.FtpServer.FileSystem.InMemory;
+using Microsoft.Extensions.DependencyInjection;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.Backups;
+using Raven.Server.Documents.PeriodicBackup;
+using Raven.Tests.Core.Utils.Entities;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Server.Documents.PeriodicBackup
+{
+    public class Ftp : CloudBackupTestBase
+    {
+        public Ftp(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [FtpFact]
+        public async Task CanUploadFile()
+        {
+            var services = new ServiceCollection();
+            services.Configure<InMemoryFileSystemOptions>(opt => opt.KeepAnonymousFileSystem = true);
+            services.AddFtpServer(builder => builder.UseInMemoryFileSystem().EnableAnonymousAuthentication());
+            services.Configure<FtpServerOptions>(opt => opt.ServerAddress = "127.0.0.1");
+            await using (var serviceProvider = services.BuildServiceProvider())
+            {
+                var ftpServerHost = serviceProvider.GetRequiredService<IFtpServerHost>();
+                await ftpServerHost.StartAsync(CancellationToken.None);
+                var settings = new FtpSettings
+                {
+                    Url = "ftp://127.0.0.1",
+                    UserName = "anonymous",
+                    Password = "itay@ravendb.net"
+                };
+                using (var client = new RavenFtpClient(settings))
+                using (var stream = new MemoryStream(Encoding.UTF8.GetBytes("abc")))
+                {
+                    client.UploadFile("testFolder", "testFile", stream);
+                    var isExist = CheckFile(settings.Url, "testFolder", "testFile", client);
+                    Assert.Equal(true, isExist);
+                }
+                await ftpServerHost.StopAsync(CancellationToken.None);
```

Comment:
can this be in some kind of a `try/finally` block?

Reply:
Done